### PR TITLE
convert rendering to use cookiecutter

### PR DESCRIPTION
**Pull request by snopoke**
This replaces the custom Jinja rendering with cookiecutter.

Notes:
* Some minor patching is done to allow the use of different template syntax in filenames vs file contents.
* The template packs are migrated to comply with what cookiecutter expects (root directory must contain a `cookiecutter` variable expression.
* `.j2` file extensions are removed
* The templates directory is excluded from `ruff` pre-commit hooks
* I didn't use the main `cookiecutter` entrypoint to avoid having to define a `cookiecutter.json` file in each template pack with all the used variables.

### CODE REVIEW

File: pegasus/monkeypatch.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    patch_find_template()
+    patch_create_env_with_context()
+
+
+def patch_find_template():
+    """Patch the Jinja env for this method to use the overlay env.
+    This is called when cookiecutter is searching for the template directory."""
+    from cookiecutter import generate
+
+    def new_find_template(repo, env, find_template=generate.find_template):
```
Comment (czue) [inline pegasus/monkeypatch.py:21]: why is `find_template` an argument here? does it have to do with preventing monkey-patching from happening more than once?

File: pegasus/templates.py [hunk 2]
```diff
@@ -1,81 +0,0 @@
-import dataclasses
```
Comment (czue) [inline pegasus/templates.py:1]: :skull:

**Review by czue — APPROVED** [review]
:+1: looks good

File: pegasus/monkeypatch.py (hunk 1 above)
Comment (snopoke) [reply pegasus/monkeypatch.py:21]: no, it's just a way of making this function a closure

**Comment by czue** [comment]
@snopoke it seems like this may have dropped the functionality that allowed you to specify a different directory for your templates and have them go there. is there an easy way to restore that?

**Comment by snopoke** [comment]
> If I change the output_dir in the `render_cookiecutter` function it is close, but seems to add an extra `/templates/` to the path. am guessing i can figure it out from here.

I've pushed a fix

**Comment by czue** [comment]
thanks! looks good now. For my own understanding of how that works - is `template_dir_name` a cookiecutter thing? or where is that used?

**Comment by czue** [comment]
Oh I see it's a cookiecutter var in the file path, which is why grep wasn't working. thanks!

**Pull request merged**